src/components/XAxis.tsx

```diff
@@ -0,0 +1,56 @@
+import { axisBottom } from 'd3-axis';
+import { select } from 'd3-selection';
+import React, { useRef } from 'react';
+
+import { usePlotContext } from '../hooks';
+import type { AxisProps } from '../types';
+
+const XAxis = ({ label, fontSize = 16, showTicks }: AxisProps) => {
+  const axisRef = useRef(null);
+  const { xScale, margin, height, width } = usePlotContext();
+  if (axisRef?.current) {
```

Comment:
This is wrong, you should never have side-effects directly in the render. this code should be in a `useEffect`

Reply:
it's even better to do it in a `useLayoutEffect` because the effect will be fired synchronously after the dom was mutated by react, and this way it will save the browser one paint, which is an intensive task

Reply:
Thank you guys